practices/graded-browser-support.md

```diff
@@ -1,67 +1,90 @@
+This approach works by using feature detection in order to determine which browsers (Grade-A and Grade-X) will receive the full experience (i.e. they "[cut the mustard](https://en.wiktionary.org/wiki/cut_the_mustard)") and which ones (Grade-C) will receive the core experience.
 
-Note: As mentioned above, while our graded browser support table specifies Internet Explorer 10 and below as Grade-C, until an updated media query is found this approach will render IE10 as Grade-A.
+### Implementation details
+
+All browser versions load a basic stylesheet that contains only [normalisation](https://necolas.github.io/normalize.css/) and a few small enhancements to the user-agent stylesheet.
+
+We then use CSS media queries to detect capable browsers. Unlike the approach described by the BBC, we don't use a JavaScript-based featured detection. We want users to have the best possible experience even if JavaScript is not available.
+
+To load the full experience only in grade-A and grade-X browsers we implement logic in the media attribute of the `<link>` element that identifies the main stylesheet, loading the stylesheet only in browsers that recognise the properties of that media query:
 
 ```html
 <link rel="stylesheet" href="your-css.css" media="only screen and (-webkit-min-device-pixel-ratio:0) and (min-color-index:0), (-ms-high-contrast: none), only all and (min--moz-device-pixel-ratio:0) and (min-resolution: 3e1dpcm)">
 ```
 
-We couple the loading of JavaScript to the loading of the enhanced CSS.
-If the browser loads the CSS within the media query, then load the JavaScript.
+This technique is documented in [Cutting the Mustard with Media queries](https://www.sitepoint.com/cutting-the-mustard-with-css-media-queries/). The specific media queries we use are based upon [CSS Only Mustard Cut](https://github.com/Fall-Back/CSS-Mustard-Cut), with a preference towards combining them into one rather than separating them out into multiple `<link>` elements. Note that you **cannot** add line breaks to the media query if they are combined.
 
-In order to check this, we use [window.matchMedia](https://developer.mozilla.org/en/docs/Web/API/Window/matchMedia).
+We couple the loading of JavaScript to the loading of the enhanced CSS. We use the [window.matchMedia](https://developer.mozilla.org/en/docs/Web/API/Window/matchMedia) API to detect when the browser loads the CSS within the media query. Once the enhanced CSS loads, this will cause the JavaScript code to load too.
 
 ```javascript
 (function() {
-    var linkEl = document.querySelector('link');
-    if (window.matchMedia && window.matchMedia(linkEl.media).matches) {
-        var script = document.createElement('script');
-        script.src = 'your-script.js';
-        script.async = true;
-        document.body.appendChild(script);
-    }
+  var linkEl = document.querySelector('link');
```

Comment:
I think we should take the chance to change this so that the appropriate <link> is specified by an ID attribute. I've seen people try to implement this when there are multiple <link>s on a page, and things have broken as a result.

Reply:
🤷 Bear in mind I don't use any of this stuff, if you provide me with the right code I can just slap it there.

Reply:
`core-stylesheet` ?